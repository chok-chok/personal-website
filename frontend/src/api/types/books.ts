export interface ListBooksQuery { 
    status?: string
};

export interface Book {
    title: string,
    author: string,
    readStatus: string,
    coverImage: string,
    review?: string,
    rating?: number,
    readDate?: Date,
    id?: string
}

