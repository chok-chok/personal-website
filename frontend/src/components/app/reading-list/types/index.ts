export interface ReadingListHeaderProps {
  headerText: string;
  bodyText: string;
}
