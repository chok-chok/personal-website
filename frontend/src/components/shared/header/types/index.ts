export interface MenuItem {
  title: string;
  url: string;
}

export interface MenuProps {
  menus: MenuItem[];
}
