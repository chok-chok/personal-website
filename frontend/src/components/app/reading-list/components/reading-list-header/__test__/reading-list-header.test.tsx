import * as React from 'react';
import ReadingListHeader from '../reading-list-header';
import { shallow } from 'enzyme';
import { ReadingListHeaderProps } from '../../../types';

const props: ReadingListHeaderProps = {
  headerText: 'Reading List',
  bodyText:
    'I love to read about various of topics. here are some selection of books which Iâ€™d like to share my rating and reviews',
};

describe('Test ReadingListHeader component', () => {
  it('should renders correctly', () => {
    const tree = shallow(<ReadingListHeader headerText={props.headerText} bodyText={props.bodyText} />);

    expect(tree).toMatchSnapshot();
  });
});
