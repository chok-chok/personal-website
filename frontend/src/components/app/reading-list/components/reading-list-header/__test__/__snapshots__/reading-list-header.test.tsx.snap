// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ReadingListHeader component should renders correctly 1`] = `
<Fragment>
  <styled__StyledH1>
    Reading List
  </styled__StyledH1>
  <styled__StyledBodyText>
    I love to read about various of topics. here are some selection of books which Iâ€™d like to share my rating and reviews
  </styled__StyledBodyText>
</Fragment>
`;
